.clang-tidy

```diff
@@ -100,8 +84,128 @@ Checks: >-
+  -clang-analyzer-core.uninitialized.UndefReturn,
+  -clang-analyzer-cplusplus.Move,
+  -clang-analyzer-cplusplus.NewDelete,
+  -clang-analyzer-cplusplus.NewDeleteLeaks,
+  -clang-analyzer-deadcode.DeadStores,
+  -clang-analyzer-nullability.NullPassedToNonnull,
+  -clang-analyzer-nullability.NullReturnedFromNonnull,
+  -clang-analyzer-nullability.NullableDereferenced,
+  -clang-analyzer-nullability.NullablePassedToNonnull,
+  -clang-analyzer-optin.cplusplus.UninitializedObject,
+  -clang-analyzer-optin.cplusplus.VirtualCall,
+  -clang-analyzer-optin.mpi.MPI-Checker,
+  -clang-analyzer-optin.osx.cocoa.localizability.EmptyLocalizationContextChecker,
+  -clang-analyzer-optin.osx.cocoa.localizability.NonLocalizedStringChecker,
+  -clang-analyzer-osx.API,
+  -clang-analyzer-osx.SecKeychainAPI,
+  -clang-analyzer-osx.cocoa.AtSync,
+  -clang-analyzer-osx.cocoa.ClassRelease,
+  -clang-analyzer-osx.cocoa.Dealloc,
+  -clang-analyzer-osx.cocoa.IncompatibleMethodTypes,
+  -clang-analyzer-osx.cocoa.NSAutoreleasePool,
+  -clang-analyzer-osx.cocoa.NSError,
+  -clang-analyzer-osx.cocoa.NilArg,
+  -clang-analyzer-osx.cocoa.ObjCGenerics,
+  -clang-analyzer-osx.cocoa.RetainCount,
+  -clang-analyzer-osx.cocoa.SelfInit,
+  -clang-analyzer-osx.cocoa.SuperDealloc,
+  -clang-analyzer-osx.cocoa.UnusedIvars,
+  -clang-analyzer-osx.cocoa.VariadicMethodTypes,
+  -clang-analyzer-osx.coreFoundation.CFError,
+  -clang-analyzer-osx.coreFoundation.CFNumber,
+  -clang-analyzer-osx.coreFoundation.CFRetainRelease,
+  -clang-analyzer-osx.coreFoundation.containers.OutOfBounds,
+  -clang-analyzer-osx.coreFoundation.containers.PointerSizedValues,
+  -clang-analyzer-security.FloatLoopCounter,
+  -clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling,
+  -clang-analyzer-security.insecureAPI.UncheckedReturn,
+  -clang-analyzer-security.insecureAPI.bcmp,
+  -clang-analyzer-security.insecureAPI.bcopy,
+  -clang-analyzer-security.insecureAPI.bzero,
+  -clang-analyzer-security.insecureAPI.getpw,
+  -clang-analyzer-security.insecureAPI.gets,
+  -clang-analyzer-security.insecureAPI.mkstemp,
+  -clang-analyzer-security.insecureAPI.mktemp,
+  -clang-analyzer-security.insecureAPI.rand,
+  -clang-analyzer-security.insecureAPI.strcpy,
+  -clang-analyzer-security.insecureAPI.vfork,
+  -clang-analyzer-unix.API,
+  -clang-analyzer-unix.Malloc,
+  -clang-analyzer-unix.MallocSizeof,
+  -clang-analyzer-unix.MismatchedDeallocator,
+  -clang-analyzer-unix.Vfork,
+  -clang-analyzer-unix.cstring.BadSizeArg,
+  -clang-analyzer-unix.cstring.NullArg,
+  -cppcoreguidelines-avoid-c-arrays,
+  -cppcoreguidelines-avoid-magic-numbers,
+  -cppcoreguidelines-c-copy-assignment-signature,
+  -cppcoreguidelines-explicit-virtual-functions,
+  -cppcoreguidelines-non-private-member-variables-in-classes,
+  -fuchsia-header-anon-namespaces,
+  -google-readability-braces-around-statements,
+  -google-readability-function-size,
+  -google-readability-namespace-comments,
+  -hicpp-avoid-c-arrays,
```

Comment:
for the hicpp checks it would be easier to use:
-hicpp*,
hicpp-no-assembler,
hicpp-exception-baseclass,

(disable all of them and only enable the 2 we are currently checking)

Reply:
I think it's better to not disable a class that may have interesting checks. This way we may get new useful tests with a new clang-tidy version even before the config is updated. The performance loss of some new superfluous aliases is less important IMO.